docs/concepts/runtime_capabilities.md

```diff
@@ -103,21 +103,85 @@ Both the [`elastic` connector](../reference/connectors/elastic.md) and the [`htt
+
+**Sink**: Depending on its setting, the `kafka_producer` connector acknowledges successful production of the event and will fail an event if it can't produce it to the configured topic.
+
+**Result**: The upstream system has no guarantees if its messages arrived anywhere. If a message is delivered to tremor it will make a best effort attempt to deliver it, but if Kafka fails to persist the event, a replay can't happen due to UDP's limitations.
+
+### Example 3: from http_server to kafka_producer
+
+```mermaid
+graph LR
+    A{http_server} -->|send| B(pipeline)
+    B -->|send| C{kafka_producer}
+```
+
+**Source**: The `http_server` connector will acknowledge successful delivery of the message to its upstream system but does not support replays. Events that did fail inside tremor trigger a `500 Internal Server Error` HTTP response.
+
+**Sink**: Depending on its setting, `kafka_producer` acknowledges successful storage of the message and will fail an event if it can't store it.
+
+**Result**: Events that are delivered to tremor will be delivered to Kafka. If the `kafka_producer` fails to persist the message, an error is sent back to the upstream system as a 500 response to the HTTP request. How the upstream system handles this failure is up to the service.
+
+### Example 4: from http_server via wal connector to kafka_producer
+
+```mermaid
+graph LR
+    A{http_server} -->|send| B(pipeline http_wal)
+    B -->|send| C{wal}
+    C -> |send| D(pipeline wal_kafka)
+    D -->|send| E{kafka_producer}
+```
+
+**Source**: The `http_server` connector will acknowledge successful delivery of the message to its upstream system but does not support replays. Events that did fail inside tremor trigger a `500 Internal Server Error` HTTP response.
+
+**Intermidiary**: This setup uses a [`wal` connector](../reference/connectors/wal.md) as an intermediary, effectively splitting the system into two pipelines. One pipeline being `http_server` to `wal` the other being `wal` to `kafka_producer`.
+
+**Sink**: Depending on its setting, `kafka_producer` acknowledges successful storage of the message and will fail an event if it can't store it.
+
+**Result 1 (http -> wal)**: The wal connector, as long as it doesn't run full, will always acknowledge messages it stores, so all data that is successfully sent to the HTTP endpoint, no matter the state of the downstream system, will be acknowledged. This frees the HTTP API from having to consider retries.
+
+**Result 2 (wal -> kafka)**: The [`wal` connector](../connectors/wal.md) acts pretty similar to the `kafka_consumer`. Whenever an event is marked as failed, it will replay all events since that one. So failure in the `kafka_producer` producing events will lead to a re-transmission from the `wal` - this happens in a transparent matter from the point of view of the `http_server` connector.
```

Comment:
```suggestion
**Result 2 (wal -> kafka)**: The [`wal` connector](../reference/connectors/wal.md) acts pretty similar to the `kafka_consumer`. Whenever an event is marked as failed, it will replay all events since that one. So failure in the `kafka_producer` producing events will lead to a re-transmission from the `wal` - this happens in a transparent matter from the point of view of the `http_server` connector.
```